File: app/schema/file_scan.py
"""
文件扫描相关的Schema定义
"""
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class ScanTriggerRequest(BaseModel):
    """手动触发扫描请求模型"""
    force_rescan: Optional[bool] = Field(False, description="是否强制重新扫描，默认False")


class ScanResultResponse(BaseModel):
    """扫描结果响应模型"""
    scan_id: int = Field(..., description="扫描记录ID")
    scan_time: datetime = Field(..., description="扫描时间")
    total_files: int = Field(..., description="扫描文件总数")
    new_found: int = Field(..., description="新发现视频数")
    already_exists: int = Field(..., description="已存在视频数")
    scan_duration: float = Field(..., description="扫描耗时(秒)")
    status: str = Field(..., description="扫描状态: running/success/failed")
    error_message: Optional[str] = Field(None, description="错误信息")

    class Config:
        from_attributes = True


class ScanRecordListResponse(BaseModel):
    """扫描记录列表响应模型"""
    records: List[ScanResultResponse] = Field(..., description="扫描记录列表")
    total: int = Field(..., description="总记录数")

    class Config:
        from_attributes = True

File: frontend/src/apis/fileScan.ts
/*
 * @Author: Await
 * @Date: 2026-01-11
 * @Description: 文件扫描 API
 */
import { request } from "../utils/requests";

// 扫描结果类型定义
export interface ScanResult {
    scan_id?: number;           // 扫描ID
    scan_time: string;          // 扫描时间
    total_files: number;        // 扫描文件总数
    new_found: number;          // 新发现视频数
    already_exists: number;     // 已存在视频数
    scan_duration: number;      // 扫描耗时（秒）
    status: 'success' | 'failed' | 'running';
    error_message?: string;     // 错误信息
}

// 扫描记录类型定义
export interface ScanRecord {
    id: number;
    scan_time: string;
    total_files: number;
    new_found: number;
    already_exists: number;
    scan_duration: number;
    status: string;
    error_message?: string;
}

// 分页响应类型
export interface PaginatedScanRecords {
    items: ScanRecord[];
    total: number;
    page: number;
    page_size: number;
}

/**
 * 触发文件扫描
 * @returns 扫描结果
 */
export async function triggerScan(): Promise<ScanResult> {
    const response = await request.request({
        url: '/file-scan/trigger',
        method: 'post'
    });
    return response.data.data;
}

/**
 * 获取扫描记录（分页）
 * @param page 页码（从1开始）
 * @param pageSize 每页数量
 * @returns 分页的扫描记录
 */
export async function getScanRecords(
    page: number = 1,
    pageSize: number = 10
): Promise<PaginatedScanRecords> {
    const response = await request.request({
        url: '/file-scan/records',
        method: 'get',
        params: { page, page_size: pageSize }
    });
    return response.data.data;
}

/**
 * 获取最新一次扫描结果
 * @returns 最新扫描结果
 */
export async function getLatestScan(): Promise<ScanResult | null> {
    const response = await request.request({
        url: '/file-scan/latest',
        method: 'get'
    });
    return response.data.data;
}

/**
 * 获取当前扫描状态
 * @returns 扫描状态信息
 */
export async function getScanStatus(): Promise<{
    is_scanning: boolean;
    current_scan_id: number | null;
}> {
    const response = await request.request({
        url: '/file-scan/status',
        method: 'get'
    });
    return response.data.data;
}
